OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -269,11 +269,12 @@ public string VoicePhraseForOrder(Actor self, Order order)
 		public bool HasSpace(int weight) { return totalWeight + reservedWeight + weight <= Info.MaxWeight; }
 		public bool IsEmpty(Actor self) { return cargo.Count == 0; }
 
-		public Actor Peek(Actor self) { return cargo.Peek(); }
+		public Actor Peek() { return cargo.Last(); }
 
-		public Actor Unload(Actor self)
+		public Actor Unload(Actor self, Actor target)
 		{
-			var a = cargo.Pop();
+			var a = cargo.First(x => x == target);
```

Comment:
I'm not sure what the intention here was: this either returns `target`, which we already know, or crashes.

This would be better implemented as 

```csharp
public Actor Unload(Actor self, Actor passenger = null)
{
    if (!cargo.Remove(passenger ?? cargo.Last()))
        return; // Passenger wasn't inside the transport

    totalWeight -= GetWeight(a);
    ... etc ...
```

and then the other overload can be removed.

Reply:
Intention of that was to be able to remove any actor inside Cargo trait in a proper manner when if it'd die inside, so that actors other than last one can be removed, though I made a weird mistake there with that lol